src/Rector/MethodCall/EloquentWhereIdToWhereKeyRector.php

```diff
@@ -0,0 +1,165 @@
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \RectorLaravel\Tests\Rector\MethodCall\EloquentWhereIdToWhereKeyRector\EloquentWhereIdToWhereKeyRectorTest
+ */
+final class EloquentWhereIdToWhereKeyRector extends AbstractRector
+{
+    public function __construct(
+        private readonly QueryBuilderAnalyzer $queryBuilderAnalyzer
+    ) {}
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Refactor model calls to the primary key using the `whereKey` and `whereKeyNot` methods',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+User::where('id', '=', $user->id)->get();
+User::where('id', $user->id)->get();
+CODE_SAMPLE,
+                    <<<'CODE_SAMPLE'
+User::whereKey($user)->get();
+User::whereKey($user)->get();
+CODE_SAMPLE
+                ),
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+User::where('id', '!=', $user->id)->get();
+User::whereNot('id', $user->id)->get();
+CODE_SAMPLE,
+                    <<<'CODE_SAMPLE'
+User::whereKeyNot($user)->get();
+User::whereKeyNot($user)->get();
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class, StaticCall::class];
+    }
+
+    /**
+     * @param  MethodCall|StaticCall  $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $node instanceof MethodCall && ! $node instanceof StaticCall) {
+            return null;
+        }
+
+        $isWhere = $this->queryBuilderAnalyzer->isMatchingCall($node, 'where');
+        $isWhereNot = $this->queryBuilderAnalyzer->isMatchingCall($node, 'whereNot');
+
+        if (! $isWhere && ! $isWhereNot) {
+            return null;
+        }
+
+        $args = $node->getArgs();
+        $argCount = count($args);
+
+        if ($argCount === 2) {
+            return $this->refactorTwoArgumentWhere($node, $args, $isWhereNot);
+        }
+
+        if ($argCount === 3 && $isWhere) {
+            return $this->refactorThreeArgumentWhere($node, $args);
+        }
+
+        return null;
+    }
+
+    /**
+     * @param  Arg[]  $args
+     */
+    private function refactorTwoArgumentWhere(MethodCall|StaticCall $node, array $args, bool $isWhereNot): ?Node
+    {
+        if (! $args[0] instanceof Arg || ! $args[0]->value instanceof String_) {
+            return null;
+        }
+
+        $columnName = $args[0]->value->value;
+        if ($columnName !== 'id') {
+            return null;
+        }
+
+        if (! $args[1] instanceof Arg || ! $args[1]->value instanceof PropertyFetch) {
+            return null;
+        }
+
+        $propertyFetch = $args[1]->value;
+        if (! $this->isName($propertyFetch->name, 'id')) {
```

Comment:
instead of hardcoding `id` here, it would be better to get a new model instance from the helpers, and then use the result of `->getKeyName()`